surfer/viz.py

```diff
@@ -2682,19 +2717,67 @@ def _min_diff(self, beg, end):
             dr = np.array(end_d['r']) - np.array(beg_d['r'])
         return (np.array(dv), dr)
 
+    def _add_scalar_data(self, data):
+        # Add scalar values to dataset
+        array_id = self._mesh_dataset.point_data.add_array(data)
+        self._mesh_dataset.point_data.get_array(array_id).name = array_id
+        self._mesh_dataset.point_data.update()
+
+        # build visualization pipeline
+        with warnings.catch_warnings(record=True):
+            pipe = mlab.pipeline.set_active_attribute(
+                self._mesh_dataset, point_scalars=array_id, figure=self._f)
+            # The new data-source is added to the wrong figure by default
+            # (a Mayavi bug??)
+            self._f.add_child(pipe.parent)
```

Comment:
Can this be fool-proofed by something like `if not pipe.parent in self._f.children` or whatever the correct check would be?